app/models/miq_worker/runner.rb

```diff
@@ -374,6 +375,32 @@ def heartbeat_to_drb
   def heartbeat_to_file(timeout = nil)
     timeout ||= worker_settings[:heartbeat_timeout] || Workers::MiqDefaults.heartbeat_timeout
     File.write(@worker.heartbeat_file, (Time.now.utc + timeout).to_s)
+
+    get_messages.each { |msg, *args| process_message(msg, *args) }
+  end
+
+  def get_messages
+    messages = []
+    now = Time.now.utc
+    @my_last_config_change ||= now
+
+    last_config_change = server_last_change(:last_config_change)
+    if last_config_change && last_config_change > @my_last_config_change
+      _log.info("#{log_prefix} Configuration has changed, New TS: #{last_config_change}, Old TS: #{@my_last_config_change}")
+      messages << ["sync_config"]
+
+      @my_last_config_change = last_config_change
+    end
+
+    messages
+  end
+
+  def key_store
+    @key_store ||= Dalli::Client.new(MiqMemcached.server_address, :namespace => "server_monitor")
+  end
```

Comment:
Since this is used in both the server and the worker models, should this be tossed into a module/mixin?

One thing I would say is we probably want to split this into three modules (don't hate me... 😅 ):

* `MiqWorkerMonitorMemcache::Connection`
* `MiqWorkerMonitorMemcache::Reader`
* `MiqWorkerMonitorMemcache::Writer`

Names are a total WIP, but the idea being that they would share the connection stuff, but probably only the `worker_monitor` would be writing to `Memcache`, while the workers would just be reading (otherwise there is mutex stuff you would have to deal with).  You might need to for other messages though, but that depends on how we implement some things.